File: 4.8-bc/ts/stage-2-feature-first/src/features/commerce/service.ts
// Commerce feature — service: замовлення.
import { randomUUID } from "node:crypto";

import type { Order } from "./model.js";
import type { Repository } from "./repository.js";

export class Service {
  constructor(private readonly repo: Repository) {}

  async place(userId: string, totalCents: number): Promise<Order> {
    const o: Order = {
      id: randomUUID(),
      userId,
      totalCents,
      status: "pending",
      createdAt: new Date(),
    };
    await this.repo.create(o);
    return o;
  }
}

File: 4.8-bc/py/stage-2-feature-first/app/features/commerce/service.py
"""Commerce feature — Place use case."""
from __future__ import annotations

from datetime import UTC, datetime
from uuid import UUID, uuid4

from app.features.commerce.model import Order
from app.features.commerce.repository import Repository


class Service:
    def __init__(self, repo: Repository) -> None:
        self._repo = repo

    async def place(self, user_id: UUID, total_cents: int) -> Order:
        o = Order(
            id=uuid4(),
            user_id=user_id,
            total_cents=total_cents,
            status="pending",
            created_at=datetime.now(UTC),
        )
        await self._repo.create(o)
        return o
